test/utils/clonableGenerator.js

```diff
@@ -0,0 +1,156 @@
+import test from 'tape';
+import { spy } from 'sinon';
+import { cloneableGenerator } from '../../src/internal/utils';
+
+test('it should proxy the original generator', assert => {
+  assert.plan(4);
+
+  const spyGenerator = {
+    args: undefined,
+    next: spy(),
```

Comment:
couldnt it be tested like rest of the code? I mean pushing to the `actual` within `next`, `throw`, `return` etc and in the end just comparing `actual` vs `expected`?